Tools/Scripts/webkitdirs.pm

```diff
@@ -1271,12 +1271,23 @@ sub determineConfiguredXcodeWorkspace()
         $configuredXcodeWorkspace = <WORKSPACE>;
         close WORKSPACE;
         chomp $configuredXcodeWorkspace;
+        return;
     }
+
+    # No configured workspace, time to find the default one.
+    # If we're using an internal SDK use the internal workspace.
+    if (xcodeSDK() =~ /\.internal$/) {
+        $configuredXcodeWorkspace = Cwd::realpath(sourceDir() . "/../Internal/Safari.xcworkspace");
+        die "using internal SDK but unable to find adjacent Internal directory: $configuredXcodeWorkspace. SDK: $xcodeSDK" unless -e $configuredXcodeWorkspace;
```

Comment:
One thing to note is that this won't abort `make` unless we also change this logic to propagate the exit code from shell backticks:

https://github.com/WebKit/WebKit/blob/3d472e998b647fa0a30124147ae30f1da1e60010/Makefile.shared#L172-L173

I don't think you need to do that now—if webkitdirs _does_ die here, the error will be printed up front and the Xcode build should fail immediately, too.

Reply:
Oh, interesting that make doesn't propagate non-zero exit status from sub-commands. As you said it's probably not a big deal.